src/lein_monolith/plugin.clj

```diff
@@ -46,27 +47,32 @@
   "Constructs a profile map containing the inherited properties from a parent
   project map."
   [monolith inherit-key setting]
-  (when-let [base-properties (get-in monolith [:monolith inherit-key])]
-    (cond
-      ; Don't inherit anything
-      (not setting)
-      nil
-
-      ; Inherit the base properties specified in the parent.
-      (true? setting)
-      (select-keys monolith base-properties)
-
-      ; Provide additional properties to inherit, or replace if metadata is set.
-      (vector? setting)
-      (->> (if (:replace (meta setting))
-             setting
-             (distinct (concat base-properties setting)))
-           (select-keys monolith))
-
-      :else
-      (throw (ex-info (str "Unknown value type for monolith inherit setting: "
-                           (pr-str setting))
-                      {:inherit setting})))))
+  (let [raw? (str/ends-with? (name inherit-key) "-raw")
+        source (if raw?
+                 (get-in monolith [:monolith :raw])
+                 monolith)
+        base-properties (get-in source [:monolith inherit-key])]
```

Comment:
Could pass in a flag to avoid the conversion to string and to allow more flexibility in choice of inherit-keys.